src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -78,32 +108,13 @@ protected static void addRepositories(SCM scm, EnvVars env, Collection<GitHubRep
      * Default implementation that looks at SCM
      */
     @Extension
-    @SuppressWarnings("unused")
     public static class FromSCM extends AbstractFromSCMImpl {
         @Override
-        public void parseAssociatedNames(AbstractProject<?, ?> job, Collection<GitHubRepositoryName> result) {
-            addRepositories(job.getScm(), buildEnv(job), result);
-        }
-    }
-
-    /**
-     * MultiSCM support separated into a different extension point since this is an optional dependency
-     */
-    @Extension(optional=true)
-    @SuppressWarnings("unused")
-    public static class FromMultiSCM extends AbstractFromSCMImpl {
-        // make this class fail to load if MultiSCM is not present
-        public FromMultiSCM() { MultiSCM.class.toString(); }
-
-        @Override
-        public void parseAssociatedNames(AbstractProject<?, ?> job, Collection<GitHubRepositoryName> result) {
-            if (job.getScm() instanceof MultiSCM) {
-                EnvVars env = buildEnv(job);
-
-                MultiSCM multiSCM = (MultiSCM) job.getScm();
-                List<SCM> scmList = multiSCM.getConfiguredSCMs();
-                for (SCM scm : scmList) {
-                    addRepositories(scm, env, result);
+        public void parseAssociatedNames(Job<?, ?> job, Collection<GitHubRepositoryName> result) {
+            SCMTriggerItem item = SCMTriggerItems.asSCMTriggerItem(job);
+            if (item != null) {
+                for (SCM scm : item.getSCMs()) {
+                    addRepositories(scm, buildEnv(job), result);
```

Comment:
Could extract the result of `buildEnv(job)` out of the loop. Rarely matters, though.